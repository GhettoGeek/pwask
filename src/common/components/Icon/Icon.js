import React from 'react'
import PropTypes from 'prop-types'
import { Icon, IconButton } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles({
  iconButton: {
    padding: 10,
  },
})

function EnhancedIcon({
  type, name, color, fontSize, onClick, ...props
}) {
  const classes = useStyles()

  if (type === 'material-ui') {
    return (
      <IconButton
        color={color}
        className={classes.iconButton}
        aria-label="icon-button"
        onClick={onClick}
      >
        <Icon fontSize={fontSize} {...props}>
          {name}
        </Icon>
      </IconButton>
    )
  }

  if (type === 'custom') {
    import(`./assets/${name}.svg`).then((IconComponent) => (
      <IconComponent />
    ))
  }

  return false
}

EnhancedIcon.propTypes = {
  type: PropTypes.oneOf(['custom', 'material-ui']),
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  fontSize: PropTypes.string,
  onClick: PropTypes.func,
}

EnhancedIcon.defaultProps = {
  type: 'material-ui',
  color: 'primary',
  fontSize: 'default',
  onClick: () => {},
}

export default EnhancedIcon
