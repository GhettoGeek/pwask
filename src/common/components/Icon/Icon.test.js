import React from 'react'
import { shallow } from 'enzyme'
import Icon from './Icon'

describe('Icon', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Icon type="material-ui" name="search" />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
