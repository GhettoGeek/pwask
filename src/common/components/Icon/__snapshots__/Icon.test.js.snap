// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly the component 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorPrimary\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"icon-button\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\">search</span></span></button>"`;
